packages/db/tests/models/test-granules-executions-model.js

```diff
@@ -0,0 +1,194 @@
+  t.context.granulesExecutionsPgModel = new GranulesExecutionsPgModel();
+
+  const collectionPgModel = new CollectionPgModel();
+  [t.context.collectionCumulusId] = await collectionPgModel.create(
+    t.context.knex,
+    fakeCollectionRecordFactory()
+  );
+
+  t.context.executionPgModel = new ExecutionPgModel();
+});
+
+test.beforeEach(async (t) => {
+  const [executionCumulusId] = await t.context.executionPgModel.create(
+    t.context.knex,
+    fakeExecutionRecordFactory()
+  );
+  t.context.executionCumulusId = executionCumulusId;
+  const [granuleCumulusId] = await t.context.granulePgModel.create(
+    t.context.knex,
+    fakeGranuleRecordFactory({
+      collection_cumulus_id: t.context.collectionCumulusId,
+    })
+  );
+  t.context.granuleCumulusId = granuleCumulusId;
+  t.context.joinRecord = {
+    execution_cumulus_id: executionCumulusId,
+    granule_cumulus_id: Number(granuleCumulusId),
+  };
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex(tableNames.granulesExecutions).where(t.context.joinRecord).del();
+});
+
+test.after.always(async (t) => {
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
+test('GranulesExecutionsPgModel.create() creates a new granule/execution join record', async (t) => {
+  const {
+    knex,
+    granulesExecutionsPgModel,
+    joinRecord,
+  } = t.context;
+
+  t.plan(1);
+
+  await knex.transaction(async (trx) => {
+    await granulesExecutionsPgModel.create(trx, joinRecord);
+    const records = await trx(tableNames.granulesExecutions).where(joinRecord);
+    t.is(
+      records.length,
+      1
+    );
+  });
+});
+
+test('GranulesExecutionsPgModel.exists() correctly returns true', async (t) => {
+  const {
+    knex,
+    granulesExecutionsPgModel,
+    joinRecord,
+  } = t.context;
+
+  t.plan(1);
+
+  await knex.transaction(async (trx) => {
+    await granulesExecutionsPgModel.create(trx, joinRecord);
+    t.true(
+      await granulesExecutionsPgModel.exists(trx, joinRecord)
+    );
+  });
+});
+
+test('GranulesExecutionsPgModel.exists() correctly returns false', async (t) => {
+  const {
+    knex,
+    granulesExecutionsPgModel,
+  } = t.context;
+
+  t.plan(1);
+
+  await knex.transaction(async (trx) => {
+    t.false(
+      await granulesExecutionsPgModel.exists(trx, {
+        execution_cumulus_id: 5,
+        granule_cumulus_id: 5,
+      })
+    );
+  });
+});
+
+test('GranulesExecutionsPgModel.upsert() creates a new granule/execution join record', async (t) => {
+  const {
+    knex,
+    granulesExecutionsPgModel,
+    joinRecord,
+  } = t.context;
+
+  t.plan(1);
+
+  await knex.transaction(async (trx) => {
+    await granulesExecutionsPgModel.upsert(trx, joinRecord);
+    const records = await trx(tableNames.granulesExecutions).where(joinRecord);
+    t.is(
+      records.length,
+      1
+    );
+  });
+});
+
+test('GranulesExecutionsPgModel.upsert() overwrites a new granule/execution join record', async (t) => {
```

Comment:
Nit: 

Interesting.  

I'm not sure what value there is in `create` not upserting by default for a join table model given the nature of that record.  Functionally it's *explicit* so that seems ok, but it might be worth style/pattern discussion.   Nothing to hold the PR up on though.

Reply:
Yeah I think the `create` was only even created for the purpose of unit tests, but then there's no reason that `upsert` couldn't be used instead....

Reply:
67506a7de1d571cc58ecceb3f662b489a0e1198d